syntax = "proto3";

package skaimsg.skaimot;

// top left bottom right defined bounding box
message BoundingBox {
    // float 0 to 1
    message xy {
        float x = 1;
        float y = 2;
    }
    xy topleft = 1;
    xy botright = 2;
}

// face embedding for improving recall/recognition
message FaceEmbedding {
    // 512 list of floats
    repeated float vals = 1;
}

// bbox embedding for improving recall/recognition
message BBoxEmbedding {
    // 2048 list of floats
    repeated float vals = 1;
}

// employee vs customer vs uninitialized classification
enum Classification {
    INIT = 0;
    EMPLOYEE = 1;
    CUSTOMER = 2;
}

message SkaimotProtoMsg {
    
    message CameraFrame {
        
        // camera mac address turned into serial num
        uint64 camera_id = 1;

        // 1e9 * epoch timestamp
        uint64 timestamp = 2;

        // people in frame
        message Person {
            
            // skaimot id num
            uint64 id = 1;

            // classification enum 
            Classification classification = 2;

            BoundingBox box = 3;
            FaceEmbedding face_embedding = 4;
            BBoxEmbedding bbox_embedding = 5;
        }
        repeated Person people_in_frame = 3;

    }
    repeated CameraFrame camera_frames = 1;

}
